packages/x-charts/src/hooks/useTicks.ts

```diff
@@ -107,77 +112,119 @@ export function useTicks(
+      return {
         value,
-        formattedValue: valueFormatter?.(value, { location: 'tick', scale }) ?? `${value}`,
+        formattedValue:
+          valueFormatter?.(value, {
+            location: 'tick',
+            scale,
+            tickNumber,
+            defaultTickLabel,
+          }) ?? defaultTickLabel,
         offset: scale(value)!,
         labelOffset: 0,
-      }));
-    }
+      };
+    });
+  }
 
-    const domain = scale.domain();
-    // Skip axis rendering if no data is available
-    // - The domains contains Infinity for continuous scales.
-    if (domain.some(isInfinity)) {
-      return [];
-    }
-    const tickLabelPlacement = tickLabelPlacementProp;
-    const ticks = typeof tickInterval === 'object' ? tickInterval : scale.ticks(tickNumber);
-
-    // Ticks inside the drawing area
-    const visibleTicks: TickItemType[] = [];
-
-    for (let i = 0; i < ticks.length; i += 1) {
-      const value = ticks[i];
-      const offset = scale(value);
-      const isInside = direction === 'x' ? instance.isXInside(offset) : instance.isYInside(offset);
-
-      if (isInside) {
-        visibleTicks.push({
-          value,
-          formattedValue:
-            valueFormatter?.(value, { location: 'tick', scale }) ??
-            scale.tickFormat(tickNumber)(value),
-          offset,
-          // Allowing the label to be placed in the middle of a continuous scale is weird.
-          // But it is useful in some cases, like funnel categories with a linear scale.
-          labelOffset:
-            tickLabelPlacement === 'middle'
-              ? scale(ticks[i - 1] ?? 0) - (offset + scale(ticks[i - 1] ?? 0)) / 2
-              : 0,
-        });
-      }
+  const domain = scale.domain();
+  // Skip axis rendering if no data is available
+  // - The domains contains Infinity for continuous scales.
+  if (domain.some(isInfinity)) {
+    return [];
+  }
+  const tickLabelPlacement = tickLabelPlacementProp;
+  const ticks = typeof tickInterval === 'object' ? tickInterval : scale.ticks(tickNumber);
+
+  // Ticks inside the drawing area
+  const visibleTicks: TickItemType[] = [];
+
+  for (let i = 0; i < ticks.length; i += 1) {
+    const value = ticks[i];
+    const offset = scale(value);
+
+    if (isInside(offset)) {
+      const defaultTickLabel = scale.tickFormat(tickNumber)(value);
+      /* If d3 returns an empty string, it means that a tick should be shown, but its label shouldn't.
+       * This is especially useful in a log scale where we want to show ticks to demonstrate it's a log
+       * scale, but don't want to show labels because they would overlap.
+       * https://github.com/mui/mui-x/issues/18239 */
+      const shouldDisplayLabel = defaultTickLabel !== '';
+
+      visibleTicks.push({
+        value,
+        formattedValue: shouldDisplayLabel
+          ? (valueFormatter?.(value, { location: 'tick', scale, tickNumber, defaultTickLabel }) ??
+            defaultTickLabel)
+          : '',
```

Comment:
Since we provide `defaultTickLabel` this logic should be on the user side

```suggestion
      visibleTicks.push({
        value,
        formattedValue: valueFormatter?.(value, { location: 'tick', scale, tickNumber, defaultTickLabel }) ??
            defaultTickLabel,
```